feat(sdk-core): add transaction validation for (consolidate) unspent management

danielpeng1 opened the pull request:
#### Description:
- Reordered + refactored transaction prebuild validation to occur after keychain retrieval, ensuring `verifyTransaction` has access to required keychain data for proper spoofing protection (for both consolidate and fanout)
- Added unit test for consolidate unspent management spoofability protection in `test/btc.ts` (for UTXO coins)
- Verifies that platform responses are validated against user parameters and covers BUILD_SIGN_SEND flow

Ticket: WP-5945

danielpeng1 committed:
test(sdk-core): add consolidate unspent management spoofability test for utxo coins

Ticket: WP-5945

danielpeng1 committed:
feat(sdk-core): add transaction verification to validate platform built transactions

Ticket: WP-5945

kamleshmugdiya reviewed (changes requested):
can do this in parallel using Promise.all

kamleshmugdiya reviewed (changes requested):
can do this in parallel using `Promise.all`

danielpeng1 committed:
fix(sdk-core): use Promise.all for concurrent processing for transaction verifying, linting fixes

Ticket: WP-5945

SimonVutov committed:
feat(sdk-coin-btc): add transaction validation for fanout unspent management

- Add test case for fanout spoofing detection in btc.ts

Ticket WP-5955

TICKET: WP-5955

danielpeng1 committed:
fix(sdk-core): add keychain info to transaction verification process, fixes unit tests

Ticket: WP-5945

danielpeng1 commented on modules/sdk-core/src/bitgo/wallet/wallet.ts:
```diff
@@ -735,6 +735,18 @@ export class Wallet implements IWallet {
+        verification: params.verification ?? {},
+        reqId,
+      });
+    }
```
Thanks, using this now

nvjsr reviewed (dismissed):
LGTM

danielpeng1 committed:
fix(sdk-core): remove filter from prebuild validation

Ticket: WP-5945

SimonVutov committed:
refactor(sdk-coin-btc): move require to top

move require getDefaultWalletKeys, toKeychainObjects to top

TICKET: WP-5955

OttoAllmendinger commented on modules/sdk-coin-btc/test/unit/btc.ts:
```diff
@@ -8,6 +8,8 @@ import { Tbtc } from '../../src';
 import { BitGoAPI } from '@bitgo/sdk-api';
 import * as utxolib from '@bitgo/utxo-lib';
 
+const { getDefaultWalletKeys, toKeychainObjects } = require('../../../bitgo/test/v2/unit/coins/utxo/util/keychains');
```
why `require` instead of import?

danielpeng1 replied:
I made a small refactor to the tests to not require these.

danielpeng1 committed:
fix(sdk-coin-btc): quick refactor for wallet key handling in BTC tests

TICKET: WP-5945

lokesh-bitgo reviewed (approved):
Looks good
